### DISCUSSION TITLE
@capacitor-firebase/messaging on Web

### DISCUSSION THREAD

**Discussion by seanaguinaga** [capawesome-team/capacitor-firebase#352] [Q&A]
Do I need to check the platform before using @capacitor-firebase/messaging vs using the web version?

I am getting an error that I have no initialized the app when trying to use @capacitor-firebase/messaging on the web

I noticed that there are not Web Instructions for this package, just wanted to confirm

This is the error

<img width="819" alt="Screenshot 2023-04-05 at 9 41 01 AM" src="https://user-images.githubusercontent.com/15254340/230147447-6d5d1825-d53c-4671-8574-45fa4e686379.png">

**Comment by robingenz** [comment]
> I noticed that there are not Web Instructions for this package, just wanted to confirm

There are a few instructions linked in the [Installation](https://github.com/capawesome-team/capacitor-firebase/tree/main/packages/messaging#installation) section:
1. https://github.com/capawesome-team/capacitor-firebase/blob/main/docs/firebase-setup.md#web
2. https://github.com/capawesome-team/capacitor-firebase/tree/main/packages/messaging#web

Seems like you forgot the setup (number 1).

I will also publish a detailed guide on how to use this plugin on the web next week. I will post it here as well.

**Reply by robingenz** [reply]
@seanaguinaga Here you can find the detailed guide on how to use this plugin on the web: https://capawesome.io/blog/the-push-notifications-guide-for-capacitor/

**Reply by seanaguinaga** [reply]
That's the thing, my app is initialized in the top level, but using react.

Does the plugin automatically try to get things from window.XXX or something?

Is there anyway to pass in an instance of firebase app or will it always just try to get it from some global scope?

**Reply by robingenz** [reply]
For now, it just uses the default `FirebaseApp` instance, see: https://github.com/capawesome-team/capacitor-firebase/blob/bfd1327257994e8f70a908b17a0e0c01e99fb7f5/packages/messaging/src/web.ts#L41

If you want to pass a specific instance, please create a feature request.

**Reply by robingenz** [reply]
But your error should have nothing to do with that. The error says you don't even have a default instance. So you should check that you initialize Firebase before you use any method of this plugin.

**Reply by seanaguinaga** [reply]
What are the env variables that your version of initializeApp is referencing?

**Reply by robingenz** [reply]
What do you mean exactly? `initializeApp(...)` must be called by the application, not the plugin.
Example: https://github.com/robingenz/ionic-capacitor-firebase-messaging-demo/blob/88e40632ef294e57d0f3a95ffcb9dc885fc88285/src/app/app.component.ts#L23
